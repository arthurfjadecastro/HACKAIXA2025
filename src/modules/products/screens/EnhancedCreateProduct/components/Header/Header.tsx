import React from 'react';
import { View, TouchableOpacity } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { Text } from '@/design-system/components';
import { styles } from '../../EnhancedCreateProduct.styles';

interface HeaderProps {
  onGoBack: () => void;
  title: string;
}

export const Header: React.FC<HeaderProps> = ({
  onGoBack,
  title,
}) => {
  return (
    <View style={styles.header}>
      <TouchableOpacity 
        onPress={onGoBack} 
        style={styles.backButton}
      >
        <Ionicons name="chevron-back" size={24} color="#005CA9" />
      </TouchableOpacity>
      <Text style={styles.headerTitle}>{title}</Text>
      <View style={styles.headerSpacer} />
    </View>
  );
};
