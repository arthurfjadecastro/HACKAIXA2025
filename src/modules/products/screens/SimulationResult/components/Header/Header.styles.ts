import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  header: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingHorizontal: 16,
    paddingVertical: 12,
    backgroundColor: '#F4F8FB',
  },
  
  backButton: {
    padding: 8,
    marginLeft: -8,
  },
  
  headerTitle: {
    fontSize: 16,
    fontWeight: '600',
    color: '#333333',
    textAlign: 'center',
    flex: 1,
  },
  
  headerSpacer: {
    width: 48,
  },
});
