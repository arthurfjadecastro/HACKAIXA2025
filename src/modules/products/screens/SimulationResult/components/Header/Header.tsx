import React from 'react';
import { View, TouchableOpacity } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { Text } from '@/design-system/components';
import { styles } from './Header.styles';

interface HeaderProps {
  onBackPress: () => void;
}

export const Header: React.FC<HeaderProps> = ({ onBackPress }) => {
  return (
    <View style={styles.header}>
      <TouchableOpacity onPress={onBackPress} style={styles.backButton}>
        <Ionicons name="chevron-back" size={24} color="#005CA9" />
      </TouchableOpacity>
      <Text style={styles.headerTitle}>Resultado da Simulação</Text>
      <View style={styles.headerSpacer} />
    </View>
  );
};
