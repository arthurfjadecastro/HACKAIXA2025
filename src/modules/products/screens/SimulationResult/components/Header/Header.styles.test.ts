import { styles } from './Header.styles';

describe('Header.styles', () => {
  it('should export styles object', () => {
    expect(styles).toBeDefined();
    expect(typeof styles).toBe('object');
  });

  it('should have header style properties', () => {
    expect(styles.header).toBeDefined();
  });

  it('should have backButton style properties', () => {
    expect(styles.backButton).toBeDefined();
  });

  it('should have consistent style structure', () => {
    expect(styles).toMatchSnapshot();
  });
});
