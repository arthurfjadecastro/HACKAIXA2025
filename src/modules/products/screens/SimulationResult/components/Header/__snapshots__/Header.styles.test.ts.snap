// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.styles should have consistent style structure 1`] = `
{
  "backButton": {
    "marginLeft": -8,
    "padding": 8,
  },
  "header": {
    "alignItems": "center",
    "backgroundColor": "#F4F8FB",
    "flexDirection": "row",
    "justifyContent": "space-between",
    "paddingHorizontal": 16,
    "paddingVertical": 12,
  },
  "headerSpacer": {
    "width": 48,
  },
  "headerTitle": {
    "color": "#333333",
    "flex": 1,
    "fontSize": 16,
    "fontWeight": "600",
    "textAlign": "center",
  },
}
`;
